data/src/main/java/com/knocklock/data/repository/NotificationRepositoryImpl.kt

```diff
@@ -0,0 +1,48 @@
+package com.knocklock.data.repository
+
+import android.content.Context
+import com.knocklock.data.mapper.toEntity
+import com.knocklock.data.mapper.toModel
+import com.knocklock.data.source.local.AppDatabase
+import com.knocklock.data.source.local.lockscreen.NotificationLocalDataSource
+import com.knocklock.domain.model.Notification
+import com.knocklock.domain.repository.NotificationRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+
+/**
+ * @Created by 김현국 2022/12/06
+ * @Time 2:17 PM
+ */
+
+class NotificationRepositoryImpl constructor(
+    context: Context
+) : NotificationRepository {
+
+    private val notificationLocalDataSource = NotificationLocalDataSource(AppDatabase.getDatabase(context))
+
+    override suspend fun insertNotification(notification: Notification) {
+        notificationLocalDataSource.insertNotification(notificationEntity = notification.toEntity())
+    }
+
+    override suspend fun deleteAllNotification() {
+        notificationLocalDataSource.deleteAllNotification()
+    }
+
+    override suspend fun deleteNotificationById(id: Int) {
+        notificationLocalDataSource.deleteNotification(id = id)
+    }
+
+    override fun getNotificationList(): Flow<List<Notification>> {
+        return flow {
+            notificationLocalDataSource.getNotificationList().collect { notificationEntityList ->
+                emit(
+                    notificationEntityList.map {
+                        it.toModel()
+                    }
+                )
+            }
```

Comment:
flow로 한번 감싼 이유가 있을가요?? 혹시 데이터를 못받는 에러와 관련있을까요?
```suggestion
        return notificationLocalDataSource.getNotificationList().map { it.toModel }
```

Reply:
```kotlin
return notificationLocalDataSource.getNotificationList().map { notificationEntityList ->
            notificationEntityList.map {
                it.toModel()
            }
        }
```

로했을때 데이터가 2개씩 반환이 되네영

Reply:
@jaeryo2357  오류였던것으로 정상작동확인했습니다